src/Orleans.Core/Statistics/IPerformanceMetrics.cs

```diff
@@ -29,60 +28,13 @@ public interface ICorePerformanceMetrics
         /// Amount of physical memory on the machine
         /// </summary>
         long? TotalPhysicalMemory { get; }
-
-        /// <summary>
-        /// the current size of the send queue (number of messages waiting to be sent). 
-        /// Only captures remote messages to other silos (not including messages to the clients).
-        /// </summary>
-        int SendQueueLength { get; }
-
-        /// <summary>
-        /// the current size of the receive queue (number of messages that arrived to this silo and 
-        /// are waiting to be dispatched). Captures both remote and local messages from other silos 
-        /// as well as from the clients.
-        /// </summary>
-        int ReceiveQueueLength { get; }
-
-        /// <summary>
-        /// total number of remote messages sent to other silos as well as to the clients.
-        /// </summary>
-        long SentMessages { get; }
-
-        /// <summary>
-        /// total number of remote received messages, from other silos as well as from the clients.
-        /// </summary>
-        long ReceivedMessages { get; }
     }
 
     /// <summary>
     /// A small set of per-silo important key performance metrics
     /// </summary>
     public interface ISiloPerformanceMetrics : ICorePerformanceMetrics
-    {   
-        /// <summary>
-        /// the current size of the receive queue (number of messages that arrived to this silo and 
-        /// are waiting to be dispatched). Captures both remote and local messages from other silos 
-        /// as well as from the clients.
-        /// </summary>
-        long RequestQueueLength { get; }
-
-        /// <summary>
-        /// number of activations on this silo
-        /// </summary>
-        int ActivationCount { get; }
-
-        /// <summary>
-        /// Number of activations on this silo that were used in the last 10 minutes 
-        /// (Note: this number may currently not be accurate if different age limits 
-        /// are used for different grain types).
-        /// </summary>
-        int RecentlyUsedActivationCount { get; }
-
-        /// <summary>
-        /// Number of currently connected clients
-        /// </summary>
-        long ClientCount { get; }
-
+    {
```

Comment:
why do we remove IClientPerformanceMetrics and keep `ISiloPerformanceMetrics ` and `ICorePerformanceMetrics ` ? 

Especially `ICorePerformanceMetrics`, all of whose metrics are covered by `SiloRuntimeStatistics` ? 

Is it because other component is using it , such as Scheduler and WorkerPool? If yes, can they use `IAppEnvironmentStatistics` and `IHostEnvironmentStatistics` instead ?

Reply:
My bad, I explained it here: https://github.com/dotnet/orleans/issues/3976#issuecomment-364274529